shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/executor/ExportSchemaConfigurationExecutor.java

```diff
@@ -0,0 +1,348 @@
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.sql.Types;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Export schema configuration executor.
+ */
+@RequiredArgsConstructor
+public final class ExportSchemaConfigurationExecutor extends AbstractShowExecutor {
+    
+    private static final String CONFIG = "config";
+    
+    private static final String COLON = ":";
+    
+    private static final String SPACE = " ";
+    
+    private static final String NEWLINE = "\n";
+    
+    private static final String COLON_SPACE = COLON + SPACE;
+    
+    private static final String COLON_NEWLINE = COLON + NEWLINE;
+    
+    private static final String INDENT = SPACE + SPACE;
+    
+    private final ExportSchemaConfigurationStatement sqlStatement;
+    
+    private final ConnectionSession connectionSession;
+    
+    @Override
+    protected List<QueryHeader> createQueryHeaders() {
+        return Arrays.asList(new QueryHeader("", "", CONFIG, CONFIG, Types.VARCHAR, "VARCHAR", 128, 0, false, false, false, false));
+    }
+    
+    @Override
+    protected MergedResult createMergedResult() {
+        String schemaName = getSchemaName();
+        ShardingSphereMetaData metaData = ProxyContext.getInstance().getMetaData(schemaName);
+        StringBuilder result = new StringBuilder();
+        result.append("schemaName").append(COLON_SPACE).append(schemaName).append(NEWLINE);
+        getDataSourcesConfig(metaData, result);
+        Optional<ShardingRuleConfiguration> ruleConfig = metaData.getRuleMetaData().getConfigurations()
+                .stream().filter(each -> each instanceof ShardingRuleConfiguration).map(each -> (ShardingRuleConfiguration) each).findAny();
+        ruleConfig.ifPresent(rule -> getRulesConfig(rule, result));
+        if (!sqlStatement.getFilePath().isPresent()) {
+            return new MultipleLocalDataMergedResult(Collections.singleton(Collections.singletonList(result.toString())));
+        }
+        
+        File outFile = new File(sqlStatement.getFilePath().get());
+        try (FileOutputStream stream = new FileOutputStream(outFile)) {
+            stream.write(result.toString().getBytes());
+            stream.flush();
+        } catch (final IOException ex) {
+            throw new ShardingSphereException(ex);
+        }
+        return null;
+    }
+    
+    private void getDataSourcesConfig(final ShardingSphereMetaData metaData, final StringBuilder result) {
+        if (null == metaData.getResource().getDataSources() || metaData.getResource().getDataSources().isEmpty()) {
+            return;
+        }
+        result.append(NEWLINE).append("dataSources").append(COLON_NEWLINE);
+        for (Map.Entry<String, DataSource> each : metaData.getResource().getDataSources().entrySet()) {
+            HikariDataSource dataSource = (HikariDataSource) each.getValue();
```

Comment:
Please use the DataSourcePropertiesCreator.create method to obtain the data source properties. It is not a good idea to cast it to HikariDataSource. It can be adjusted in the next PR.